### PR TITLE
feat(cockpit): queued-prompt UI above the TUI composer

### LINKED ISSUE #1700: TUI cockpit: queued-prompt UI above the composer
**Note:** Claude handled the investigation and prose via back-and-forth. Idea, decisions, and everything else are mine. — @Seluj78

Split out from the cockpit-in-TUI follow-ups epic #1113.

## Problem

The TUI cockpit has no queued-prompt UI. The web composer lets you queue prompts while a turn is running, shows them as an editable strip above the composer, and drains them when the agent goes idle. The TUI offers none of this.

## What's there today

The web queue is entirely client-side state, not a daemon-published event stream. This matters: the epic checklist said "daemon already publishes the events", but a grep of `src/cockpit/state.rs` finds no queue event, and the web reducer owns the whole lifecycle:

- `web/src/hooks/useCockpit.ts:47` defines the queue actions `enqueue_prompt` / `dequeue_prompt` / `edit_queued_prompt` / `clear_queue`.
- `web/src/hooks/useCockpit.ts:389` is the reducer that mutates `state.queuedPrompts`.
- `web/src/hooks/useCockpit.ts:329` drains the queue when the agent transitions to idle, honoring the `cockpit.queue_drain_mode` setting (combined vs sequential).

The web strip UI:

- `web/src/components/cockpit/CockpitView.tsx:417` renders `QueuedPromptsStrip` from `state.queuedPrompts` with remove / edit / clear handlers.
- `web/src/components/cockpit/QueuedPromptsStrip.test.tsx` pins the `QueuedPrompt` shape `{ id, text, queuedAt }`.

The TUI has no equivalent:

- `src/tui/cockpit_view/reducer.rs` stores no queued-prompt state.
- `src/tui/cockpit_view/input.rs:29` submits on Enter with no queue branch.

## Proposal

Replicate the web's client-side approach in the TUI. Add a `queued_prompts` field to the cockpit view state, queue on Enter when a turn is active, render an editable strip above the composer in `render.rs`, and drain on turn-end honoring `cockpit.queue_drain_mode`. No new daemon event is needed; this is local composer state plus a turn-end drain.

Touch points: `src/tui/cockpit_view/state.rs` (or `reducer.rs`), `src/tui/cockpit_view/input.rs`, `src/tui/cockpit_view/render.rs`. Confirm the `queue_drain_mode` setting is reachable from the TUI cockpit path.

## User stories / regression tests

Implementation MUST add tests covering each story below; PR is not done without them.

- As a cockpit TUI user, when I press Enter while a turn is active, then the prompt is queued and shown in a strip above the composer rather than submitted immediately.
  - Suggested test layer: `src/tui/cockpit_view` unit test plus `tests/e2e/`.
- As a cockpit TUI user, when the agent goes idle with a non-empty queue, then queued prompts drain per the configured `queue_drain_mode`.
  - Suggested test layer: `src/tui/cockpit_view` unit test on the drain transition.

## Open questions

- Match the web's `combined` vs sequential drain modes exactly, or start with one?
- In-place edit of queued rows in the TUI, or remove-and-retype only for v1?
- Queue navigation / focus keys?

## Out of scope

- Server-side queue persistence (queue stays client-side, as on web).

## Cross-references

- Parent epic: #1113.

### PR DESCRIPTION
**Pull request by Seluj78**
**Note:** Claude handled the implementation and prose via back-and-forth. Idea, decisions, and everything else are mine. — @Seluj78

## Description

The TUI cockpit had no queued-prompt UI: pressing Enter mid-turn either did nothing useful or fired a prompt into a running turn. The web composer has had a client-side prompt queue for a while (queue while busy, drain on idle per `cockpit.queue_drain_mode`); this brings the same behavior to the TUI cockpit.

What changed:

- Pressing Enter while the agent is busy (turn active, a POST in flight, or the WebSocket down) now parks the prompt in a local queue rendered as a **Queued (N)** strip above the composer instead of sending it. When the agent goes idle (the `Stopped` edge), the queue drains per the daemon's `cockpit.queue_drain_mode`: `combined` joins the queued entries into one follow-up, `serial` fires the head and the next idle fires the rest. The mode is read from `GET /api/about` rather than local config, because the TUI cockpit can attach to a remote daemon.
- An optimistic in-flight lock closes the POST-then-echo race so hammering Enter never double-fires a concurrent prompt. A drained batch stays in the queue until its POST succeeds, so a failed send is recoverable (next turn-end edge, a reconnect, or an empty-composer flush) rather than silently dropped.
- `Ctrl+X` clears the whole queue. Combined mode slices batches at `/clear` and `/new` boundaries so a clear command never glues to adjacent prose and corrupts slash-command parsing (matching the web's #1356 behavior).

Turn-active state is derived purely from daemon events on the reducer (so it rebuilds correctly after a `/replay`), while the queue and the in-flight lock live on the view state.

Closes #1700

## PR Type

- [x] New Feature
- [ ] Bug Fix
- [ ] Refactor
- [ ] Documentation
- [ ] Infrastructure / CI

## How to test

- [ ] Open a cockpit session in the TUI (`aoe cockpit attach <id>`), send a prompt, and while the agent is still working type another and press Enter: it appears in the **Queued (N)** strip above the composer instead of being sent.
- [ ] Queue two or more prompts during a turn; when the agent stops, confirm they drain per the configured `cockpit.queue_drain_mode` (combined: one joined follow-up; serial: one at a time across successive turns).
- [ ] With prompts queued, press `Ctrl+X` and confirm the strip clears.
- [ ] Queue a `/clear` followed by a normal prompt in combined mode and confirm `/clear` fires on its own turn, not glued to the following text.
- [ ] Hammer Enter rapidly right after a send and confirm only one prompt reaches the agent (no duplicate concurrent turn).

## Checklist

- [x] New and existing tests pass
- [x] Documentation was updated where necessary
- [ ] For UI changes: included screenshot or recording

## Test Coverage Analysis

**Web dashboard / cockpit (Playwright user story)**
- [x] N/A: this PR doesn't change a user-facing dashboard flow

**TUI / CLI (e2e test)**
- [x] Skipped a test, because: the cockpit-view logic is covered by unit tests (`queue.rs` drain batching, reducer `turn_active` edges, `is_busy()` gating, strip-height rendering). A full e2e for the queue would need the harness to drive a mid-turn cockpit view, which requires a daemon plus a fake ACP agent streaming and staying mid-turn; the current `tests/e2e/` harness only drives the home TUI with a fake `claude`. Deferred to a follow-up that adds that fake-agent terminal plumbing.

## AI Usage

- [x] AI was used

**AI Model/Tool used:** Claude (via Claude Code, agent-of-empires `/aoe-implement` skill)

**Any Additional AI Details you'd like to share:**
Investigation, plan (pressure-tested with a multi-model design debate), and implementation were drafted by Claude under my direction. I made the design calls (scope, queue semantics, drain-failure handling) and reviewed each commit.

- [x] I am an AI Agent filling out this form (check box if true)

This PR adds a client-side queued-prompt UI and plumbing to the TUI cockpit so users can park prompts while the agent is busy and have them sent automatically when the agent becomes idle.

What changed (high-level)
- UI/UX: introduces a "Queued (N)" strip above the composer that previews queued prompts; Enter is turn-aware (sends when idle, queues when busy); Shift+Enter still inserts newlines; Ctrl+X clears the queue; Enter on an empty composer can manually drain the queue when idle.
- Behavior: queue drain behavior follows the daemon-configured cockpit.queue_drain_mode (fetched from GET /api/about): "combined" batches queued entries into a follow-up, "serial" sends them one at a time when idle. Queued batches remain until their POST succeeds so failures are recoverable.
- Reliability: adds an optimistic in-flight lock to avoid duplicate concurrent POSTs when Enter is hammered; view state resets the lock on relevant daemon events (turn end, reconnect, lagged).
- State & rendering: queue, drain mode, and in_flight lock live in CockpitViewState; reducer tracks turn_active on the transcript; render now uses a 4-pane layout and draws the queued-strip with collapsed-whitespace previews and truncation.
- Tests & docs: unit tests cover queue logic and turn_active transitions; docs updated with TUI queue semantics and keybind changes.

Technical summary (concise)
- New module: PromptQueue (FIFO) with methods push, clear, iter, len, next_batch(mode) and drop_front; next_batch implements combined vs serial batching and respects /clear and /new boundaries.
- View state: CockpitViewState gains queue: PromptQueue, drain_mode: QueueDrainMode (fetched from HttpClient::queue_drain_mode()), and in_flight: bool; adds is_busy().
- Reducer: CockpitTranscript adds pub turn_active: bool and unit tests for state transitions.
- Input: Intent::ClearQueue added; Ctrl+X routed to clear queue; Enter logic updated to queue/send/drain based on is_busy().
- Networking: HttpClient::queue_drain_mode() reads cockpit_queue_drain_mode from GET /api/about.
- Rendering: queued-strip preview with whitespace collapsing, truncation, overflow (+N more) handling and height calculation.
- Error recovery: drained batches are removed only after successful POST; failed sends leave queue intact for retry on next idle/reconnect/flush.

Benefits
- Prevents user input loss while the agent is busy and improves UX parity with the web cockpit.
- Configurable drain behavior to match remote daemon settings.
- Safer concurrency: avoids accidental duplicate requests and allows recoverable retries on failures.
- Clear visual feedback and straightforward controls (queued preview, Ctrl+X).

Potential areas for follow-up
- Add e2e tests for full enqueue/drain flows and network failure recovery (some e2e tests deferred).
- Consider in-place editing of queued entries in the TUI (currently not supported).
- Review UX of combined batching with slash-commands for edge-case command composition.

### CODE REVIEW

File: src/cockpit/client/http.rs [hunk 1]
```diff
@@ -19,6 +19,14 @@ use crate::cockpit::protocol::{
 
 const DEFAULT_TIMEOUT: Duration = Duration::from_secs(15);
 
+/// Subset of the daemon's `GET /api/about` payload the cockpit client
+/// reads. The full `ServerAbout` carries many more fields; serde drops
+/// the rest.
+#[derive(serde::Deserialize)]
+struct AboutResponse {
+    cockpit_queue_drain_mode: String,
+}
```
[the remark this answers is not in the document]
Comment (Seluj78) [reply src/cockpit/client/http.rs:29]: Done in https://github.com/agent-of-empires/agent-of-empires/pull/1758/commits/78725e8c. Added `#[serde(default)]` so a missing `cockpit_queue_drain_mode` deserializes to `""`, parses to `None`, and falls back to `Combined` with no error toast.

**Comment by Seluj78** [comment]
@coderabbitai addressed the 3 `render.rs` nitpicks in https://github.com/agent-of-empires/agent-of-empires/pull/1758/commits/b5e649b4:

- Module doc updated to "Four-pane" (transcript / status / queued-prompts strip / composer).
- Queue panel now uses `BorderType::Rounded` + `Padding::horizontal(1)` per the house style.
- Queued-prompt previews collapse whitespace so multi-line entries render as one predictably-truncated line.
